### ISSUE TITLE
web2 client silently fails if given unicode address

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3276]
|<img alt="htj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| htj reported|
|-|-|
|Trac ID|trac#3276|
|Type|defect|
|Created|2008-06-04 13:14:48Z|

If the web2 client is given a unicode string as resource, the request will fail silently.

I'm not sure if this is a twisted problem or if it is only related to web2. Anyway, silent errors are bad (and very annoying to debug).

<details><summary>Searchable metadata</summary>

```
trac-id__3276 3276
type__defect defect
reporter__htj htj
priority__normal normal
milestone__Web2_Gold_Master Web2-Gold-Master
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__ 
time__1212585288000000 1212585288000000
changetime__1297619914000000 1297619914000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

Not going to fix this in web2 (see [#4821](https://github.com/twisted/twisted/issues/4821)).  `twisted.web.client.Agent` uses `twisted.web.http.urlparse` which rejects unicode, so no need to retarget this ticket at the web component.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
